tools/rosgraph/src/rosgraph/roslogging.py

```diff
@@ -63,12 +63,18 @@ def findCaller(self, *args, **kwargs):
             f = f.f_back
         while hasattr(f, "f_code"):
             # Search for the right frame using the data already found by parent class.
-            co = f.f_code
-            filename = os.path.normcase(co.co_filename)
-            if filename == file_name and f.f_lineno == lineno and co.co_name == func_name:
-                break
+            if sys.version_info.major > 3 or (sys.version_info.major == 3 and sys.version_info.minor >= 11):
+                if f.f_code.co_name == '_base_logger':
+                    break
```

Comment:
Why is this change necessary? Where does `_base_logger` come from?

Reply:
It is the function name from here: https://github.com/ros/ros_comm/blob/noetic-devel/clients/rospy/src/rospy/core.py#L158 also already used here: https://github.com/ros/ros_comm/blob/noetic-devel/tools/rosgraph/src/rosgraph/roslogging.py#L74